shared/src/store/action-reducers/simulation.ts

```diff
@@ -346,4 +399,102 @@ export namespace SimulationActionReducers {
             },
             rights: 'trainer',
         };
+
+    export const changeAutomaticDistributionLimit: ActionReducer<ChangeAutomaticDistributionLimitAction> =
+        {
+            action: ChangeAutomaticDistributionLimitAction,
+            reducer(
+                draftState,
+                { simulatedRegionId, behaviorId, vehicleType, newLimit }
+            ) {
+                const automaticDistributionBehaviorState = getBehaviorById(
+                    draftState,
+                    simulatedRegionId,
+                    behaviorId,
+                    'automaticallyDistributeVehiclesBehavior'
+                );
+
+                automaticDistributionBehaviorState.distributionLimits[
+                    vehicleType
+                ] = newLimit;
+
+                if (newLimit < 1) {
+                    delete automaticDistributionBehaviorState.remainingInNeed[
+                        vehicleType
+                    ];
+                } else {
+                    if (
+                        !automaticDistributionBehaviorState.remainingInNeed[
+                            vehicleType
+                        ]
+                    ) {
+                        automaticDistributionBehaviorState.remainingInNeed[
+                            vehicleType
+                        ] = cloneDeepMutable(
+                            automaticDistributionBehaviorState.distributionDestinations
+                        );
+                    }
+                }
+
+                return draftState;
+            },
+            rights: 'trainer',
+        };
+
+    export const addAutomaticDistributionLimit: ActionReducer<AddAutomaticDistributionDestinationAction> =
+        {
+            action: AddAutomaticDistributionDestinationAction,
+            reducer(
+                draftState,
+                { simulatedRegionId, behaviorId, destinationId }
+            ) {
+                const automaticDistributionBehaviorState = getBehaviorById(
+                    draftState,
+                    simulatedRegionId,
+                    behaviorId,
+                    'automaticallyDistributeVehiclesBehavior'
+                );
+
+                automaticDistributionBehaviorState.distributionDestinations[
+                    destinationId
+                ] = true;
+
+                Object.values(
+                    automaticDistributionBehaviorState.remainingInNeed
+                ).forEach((regionsInNeed) => {
+                    regionsInNeed[destinationId] = true;
+                });
```

Comment:
Should this be executed when the limit is 0?

Reply:
yes, so that if the limit ever is lifted they are marked as in need